src/main/java/de/unijena/cheminf/clustering/desccalc/Descriptor.java

```diff
@@ -920,7 +1997,986 @@ private static synchronized void setWienerNumber(
+     */
+    private static synchronized void setBCUT(
+            IAtomContainer anAtomContainer,
+            float[] aVector,
+            int aStartIndex
+    ) {
+        DoubleArrayResult result = (DoubleArrayResult) descriptorToCdkObjectMap.get(BCUT).calculate(anAtomContainer).getValue();
+        for (int i = 0; i < 6; i++) {
+            aVector[aStartIndex + i] = (float) result.get(i);
+        }
+    }
+
+    /**
+     * Sets bond count.
+     * Note: Checks are NOT performed here. All necessary checks have already been made in public methods above.
+     * Note: Method has to be synchronized due to missing thread-safety of the CDK calculation
+     *
+     * @param anAtomContainer Molecule (IS NOT CHANGED)
+     * @param aVector Vector of molecule to be filled with calculated components of descriptors (MAY BE CHANGED)
+     * @param aStartIndex Start index in aVector to be filled with calculated components of descriptors
+     */
+    private static synchronized void setBondCountAll(
+            IAtomContainer anAtomContainer,
+            float[] aVector,
+            int aStartIndex
+    ) {
+        aVector[aStartIndex] = (float) ((IntegerResult) descriptorToCdkObjectMap.get(BOND_COUNT_ALL).calculate(anAtomContainer).getValue()).intValue();
+    }
+
+    /**
+     * Sets specific bond counts (single, double, triple).
+     * Note: Checks are NOT performed here. All necessary checks have already been made in public methods above.
+     * Note: Method has to be synchronized due to missing thread-safety of the CDK calculation
+     *
+     * @param anAtomContainer Molecule (IS NOT CHANGED)
+     * @param aVector Vector of molecule to be filled with calculated components of descriptors (MAY BE CHANGED)
+     * @param aStartIndex Start index in aVector to be filled with calculated components of descriptors
+     */
+    private static synchronized void setBondCountSpecified(
+            IAtomContainer anAtomContainer,
+            float[] aVector,
+            int aStartIndex
+    ) {
+        try {
+            BondCountDescriptor bondCountDesc = (BondCountDescriptor)descriptorToCdkObjectMap.get(BOND_COUNT_SPECIFIED);
+            bondCountDesc.setParameters(new Object[]{"s"});
+            aVector[aStartIndex] = (float) ((IntegerResult) bondCountDesc.calculate(anAtomContainer).getValue()).intValue();
+
+            // Double bonds (d)
+            bondCountDesc.setParameters(new Object[]{"d"});
+            aVector[aStartIndex + 1] = (float) ((IntegerResult) bondCountDesc.calculate(anAtomContainer).getValue()).intValue();
+
+            // Triple bonds (t)
+            bondCountDesc.setParameters(new Object[]{"t"});
+            aVector[aStartIndex + 2] = (float) ((IntegerResult) bondCountDesc.calculate(anAtomContainer).getValue()).intValue();
+        } catch (Exception e) {
+            for (int i = 0; i < 3; i++) {
+                aVector[aStartIndex + i] = Float.NaN;
```

Comment:
idea to discuss: all these methods return void right now. Maybe we could change that so that they instead return the matrix positions that they marked as NaN if calculation failed.